cmd/upgrade.go

```diff
@@ -70,6 +72,7 @@ func newUpgradeCmd() *cobra.Command {
 	f.StringVar(&uc.deploymentDirectory, "deployment-dir", "", "the location of the output from `generate`")
 	f.StringVarP(&uc.upgradeVersion, "upgrade-version", "k", "", "desired kubernetes version (required)")
 	f.IntVar(&uc.timeoutInMinutes, "vm-timeout", -1, "how long to wait for each vm to be upgraded in minutes")
+	f.IntVar(&uc.cordonDrainTimeoutInMinutes, "cordon-drain-timeout", 20, "how long to wait for each vm to be cordoned in minutes")
```

Comment:
let's move the default logic to the upgrader code (similar to how timeoutInMinutes is doing it) because AKS also calls upgrader so we want to make sure that the timeout isn't nil for them.

In other words, let's set the arg default to -1 and only set `uc.cordonDrainTimeout` if the timeout `!= -1`. Then in the upgrader, do something like https://github.com/Azure/aks-engine/blob/master/pkg/operations/kubernetesupgrade/upgrader.go#L261 and https://github.com/Azure/aks-engine/blob/master/pkg/operations/kubernetesupgrade/upgrader.go#L116.

We could arguably leave the `20` default here as well instead of setting it to `-1` but having a default timeout const in one place will make it easier to change the default in the future.

Reply:
Yeah, I was wondering why the default was being set there instead of here for step vm upgrade timeout. Now I know 😊 I’ll make the change.

Reply:
I've made the changes to set the default value in upgrader.